src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/TypeLoaderExports.cs

```diff
@@ -123,6 +124,40 @@ public static object GenericLookupAndCast(object arg, IntPtr context, IntPtr sig
             return RawCalliHelper.Call<object>(v._result, arg, v._auxResult);
         }
 
+        public static unsafe IntPtr InterfaceLookupForSlot(object obj, IntPtr slot)
+        {
+            // TODO: do something more optimal for monomorphic case
+
+            if (TryGetFromCache((IntPtr)obj.GetMethodTable(), slot, out var v))
+                return v._result;
+
+            return InterfaceLookupForSlotSlow(obj, slot);
+        }
+
+        private static unsafe IntPtr InterfaceLookupForSlotSlow(object obj, IntPtr slot)
+        {
+            // TODO: this is wrong for IDynIntfCastable
```

Comment:
Perhaps just do not cache the lookup result for `IDynIntfCastable` and only update the monomorphic when there is a cache hit in the larger cache.  

It would mean that `IDynIntfCastable` will always result in 2 misses - in the monomorphic check and then in a larger cache and then will have to resolve. However non-`IDynIntfCastable` case will be fast.